src/LazyOperations/CartesianProduct.jl

```diff
@@ -405,6 +405,41 @@ function σ(d::AbstractVector{N}, cpa::CartesianProductArray{N}) where {N<:Real}
     return svec
 end
 
+# faster version for sparse vectors
+function σ(d::AbstractSparseVector{N}, cpa::CartesianProductArray{N}
+          ) where {N<:Real}
+    svec = similar(d)
+    indices, _ = SparseArrays.findnz(d)
+    if isempty(indices)
+        # direction is the zero vector
+        return an_element(cpa)
+    end
+    next_idx = 1
+    next_dim = indices[next_idx]
```

Comment:
```suggestion
    next_dim = indices[next_idx] # chosen such that the next _dim points to a different block than the current one
```
or some comment along these lines that helps to understand / remember how the algorithm works. this variable does not correspond to a dimension, as the name "next dimension" suggests, since is comparable to the right-end `i1`, and dimensions are `i1 - i0 + 1`, but to an index that has been identified to belong to some block (and `next_dim` are chosen such that they belong to different blocks).

Reply:
.. or maybe call the variable `next_blk` for "index in the next block"